src/dag/chunk.rs

```diff
@@ -1,8 +1,100 @@
-pub type Key = String;
+use super::meta_generated;
+use flatbuffers;
+use flatbuffers::FlatBufferBuilder;
 
 #[allow(dead_code)]
+#[derive(Debug)]
 pub struct Chunk {
-    pub key: Key,
-    buf: Vec<u8>,
-    refs: [Key],
+    hash: String,
+    data: Vec<u8>,
+    meta: Option<(Vec<u8>, usize)>,
+}
+
+#[allow(dead_code)]
+impl Chunk {
```

Comment:
This might be an alternate place to put a ~3-line description of what a Chunk logically represents and how `refs` relates to it. It would help me. :)

Reply:
I added some comments to the module explaining the overall approach, plus here, let me know if it works for you.